axelor-stock/src/main/java/com/axelor/apps/stock/service/MassStockMoveServiceImpl.java

```diff
@@ -0,0 +1,177 @@
+
+  @Inject
+  public MassStockMoveServiceImpl(
+      MassStockMoveRepository massStockMoveRepository,
+      SequenceService sequenceService,
+      PickedProductRepository pickedProductRepo,
+      StockLocationLineRepository stockLocationLineRepository,
+      PickedProductService pickedProductService,
+      StoredProductService storedProductService) {
+    this.massStockMoveRepository = massStockMoveRepository;
+    this.sequenceService = sequenceService;
+    this.pickedProductRepo = pickedProductRepo;
+    this.stockLocationLineRepository = stockLocationLineRepository;
+    this.pickedProductService = pickedProductService;
+    this.storedProductService = storedProductService;
+  }
+
+  @Override
+  public void realizePicking(MassStockMove massStockMove) {
+    for (PickedProduct pickedProduct : massStockMove.getPickedProductList()) {
+      try {
+        pickedProductService.createStockMoveAndStockMoveLine(massStockMove, pickedProduct);
+      } catch (AxelorException e) {
+        TraceBackService.trace(e);
+      }
+    }
+  }
+
+  @Override
+  public void realizeStorage(MassStockMove massStockMove) {
+    for (StoredProduct storedProduct : massStockMove.getStoredProductList()) {
+      try {
+        storedProductService.createStockMoveAndStockMoveLine(storedProduct);
+      } catch (AxelorException e) {
+        TraceBackService.trace(e);
+      }
+    }
+  }
+
+  @Override
+  public void cancelPicking(MassStockMove massStockMove) {
+    for (PickedProduct pickedProduct : massStockMove.getPickedProductList()) {
+      try {
+        pickedProductService.cancelStockMoveAndStockMoveLine(massStockMove, pickedProduct);
+      } catch (AxelorException e) {
+        TraceBackService.trace(e);
+      }
+    }
+  }
+
+  @Override
+  public void cancelStorage(MassStockMove massStockMove) {
+    for (StoredProduct storedProduct : massStockMove.getStoredProductList()) {
+      try {
+        storedProductService.cancelStockMoveAndStockMoveLine(storedProduct);
+      } catch (AxelorException e) {
+        TraceBackService.trace(e);
+      }
+    }
+  }
+
+  @Override
+  @Transactional(rollbackOn = Exception.class)
+  public void importProductFromStockLocation(MassStockMove massStockMove) throws AxelorException {
+    List<StockLocationLine> stockLocationLineList =
+        stockLocationLineRepository
+            .all()
+            .filter("self.stockLocation = ?1", massStockMove.getCommonFromStockLocation())
+            .fetch();
+    List<StockLocationLine> detailsStockLocationLineList =
+        stockLocationLineRepository
+            .all()
+            .filter("self.detailsStockLocation =?1", massStockMove.getCommonFromStockLocation())
+            .fetch();
+    if (stockLocationLineList.isEmpty() && detailsStockLocationLineList.isEmpty()) {
+      throw new AxelorException(
+          TraceBackRepository.CATEGORY_NO_VALUE,
+          I18n.get("No stock location lines have been found in the common from stock location."));
```

Comment:
Every message exception should be in *Module*ExceptionMessage class. In this case StockExceptionMessage